### PR TITLE
VR Starter Tutorial rewrite/refactor

### LINKED ISSUE #2727: Update VR starter tutorial for Godot 3.1
https://docs.godotengine.org/en/latest/tutorials/vr/vr_starter_tutorial.html

The tutorial and demo project should be updated for Godot 3.1. I notice some problems running it in 3.0. When I try to run the demo project, I get this error in the debugger:

```
E 0:00:02:0992   Condition ' p_tracker_id == 0 ' is true. returned: __null
  <C Source>     servers/arvr_server.cpp:305 @ find_by_type_and_id()
```

Sometimes I get random crashes with the below error, with no other info printed. Sometimes it crashes immediately, sometimes if I leave my headset on the floor then it runs fine but then it crashes immediately when I put on my VR headset (note: it says Unix but I'm on Windows w/ a Vive).

```
ERROR: _get_socket_error: Socket error: 10054
   At: drivers/unix/net_socket_posix.cpp:190
```

Also, trying to open `addons/godot-openvr/scenes/ovr_first_person.tscn` fails because it's looking for a file in `addons/vr-common` which does not exist in the demo project.

Also, instead of a ZIP file, perhaps the demo project should be in [the demo projects repo](https://github.com/godotengine/godot-demo-projects).

Pinging @TwistedTwigleg and @BastiaanOlij

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
This is a work-in-progress pull request that updates the VR starter tutorial to Godot 3.1, while also including some improvements and additional features.

For a list of changes to the VR starter tutorial project, see [this PR](https://github.com/GodotVR/godot_openvr_fps/pull/3) on the OpenVR repository.

Closes #2727

### CODE REVIEW

**Comment by TwistedTwigleg** [comment]
The first half of the tutorial is finished! Now all that is left is rewriting/refactoring the code for the special objects, like the pistol and sword.

Also, I decided to break the tutorial into two parts and moved it to its own sub-category off the VR category. That way it should be a little easier to manage, maintain, and read.

**Comment by TwistedTwigleg** [comment]
Okay, the tutorial rewrite/refactor is finished! Everything is up to date with the project in the Godot OpenVR repository. This PR can now be reviewed and pulled in, assuming everything works and looks okay.

The OpenVR repository [PR](https://github.com/GodotVR/godot_openvr_fps/pull/3) needs to be pulled in and a release needs to be made containing the starter and finished projects. I can merge the OpenVR pull request and make a release there containing the starter and finished projects if you guys want, or I can wait until this PR is reviewed in case additional changes are necessary. Either way is cool with me.

**Comment by TwistedTwigleg** [comment]
@BastiaanOlij - Once you have a chance, can you test the project? Another pair of eyes probably couldn't hurt, and especially since the project will (potentially) be used for the Godot documentation.

**Comment by BastiaanOlij** [comment]
Will definitely do but might not get to it until I'm back in Sydney, currently visiting my folks in Holland :0

**Comment by TwistedTwigleg** [comment]
No problem, there is no rush.
Enjoy being with your family, that is more important than this PR. Take all the time you need 🙂

File: tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst [hunk 1]
```diff
@@ -0,0 +1,1146 @@
...
+            get_viewport().arvr = true
+            get_viewport().hdr = false
+
+            OS.vsync_enabled = false
+            Engine.target_fps = 90
+            # Also, the physics FPS in the project settings is also 90 FPS. This makes the physics
+            # run at the same frame rate as the display, which makes things look smoother in VR!
+
+ .. code-tab:: csharp
+
```
Comment (BastiaanOlij) [inline tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst]: C#! Love it !:)

File: tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst [hunk 2]
```diff
@@ -0,0 +1,1146 @@
...
+.. note:: The OpenVR VR interface is not included with Godot by default. You will need to download the OpenVR asset from the
+          `Asset Library <https://godotengine.org/asset-library/asset/150>`_ or `GitHub <https://github.com/GodotVR/godot-openvr-asset>`_.
+
+The code then combines two conditionals, one to check if the `VR` variable is NOT null (``if VR``) and another calls the initialize function, which returns a boolean based on
+whether the OpenVR interface was able to initialize or not. If both of these conditionals return true, then we can turn the main Godot :ref:`Viewport <class_Viewport>` into
+an ARVR viewport.
+
+If the VR interface initialized successfully, we then get the root :ref:`Viewport <class_Viewport>` and set the `arvr` property to ``true``. This will tell Godot to use the initialized
+ARVR interface to drive the :ref:`Viewport <class_Viewport>` display. After setting the ``arvr`` property to ``true``, we set the ``hdr`` property to ``false``. We do this because
```
Comment (BastiaanOlij) [inline tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst]: I think it's worth adding a note here that a version that does support HDR is available on the master repo. I'm planning on updating the version in the asset library once Godot 3.2 has been released

File: tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst [hunk 3]
```diff
@@ -0,0 +1,1146 @@
...
+          - The sky panorama was created by `CGTuts <https://cgi.tutsplus.com/articles/freebie-8-awesome-ocean-hdris--cg-5684>`_.
+          
+          - The font used is Titillium-Regular 
+          - - The font is licensed under the SIL Open Font License, Version 1.1
+          
+          - The audio used are from several different sources, all downloaded from the Sonniss #GameAudioGDC Bundle (`License PDF <https://sonniss.com/gdc-bundle-license/>`_) 
+          - - The folders where the audio files are stored have the same name as folders in the Sonniss audio bundle.
+          
+          - The OpenVR addon was created by `Bastiaan Olij <https://github.com/BastiaanOlij>`_ and is released under the MIT license. It can be found both on the `Godot Asset Library <https://godotengine.org/asset-library/asset/150>`_ and on `GitHub <https://github.com/GodotVR/godot-openvr-asset>`_.
+
```
Comment (BastiaanOlij) [inline tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst]: While my code is MIT, the library Valve supplies isn't. We may need to make a note of that.

File: tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst [hunk 4]
```diff
@@ -0,0 +1,1146 @@
...
+How the VR controller scene is setup
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In both scenes the root node is a ARVRController node. The only difference is that the ``Left_Controller`` scene has the ``Controller Id`` property set to ``1`` while
+the ``Right_Controller`` has the ``Controller Id`` property set to ``2``.
+
+Next is the ``Hand`` :ref:`MeshInstance <class_MeshInstance>` node. This node is used to display the hand mesh that will be used when the VR controller is not holding onto a
+:ref:`RigidBody <class_RigidBody>` node. The hand in the ``Left_Controller`` scene is a left hand, while the hand on the ``Right_Controller`` scene is a right hand.
+
```
Comment (BastiaanOlij) [inline tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.rst]: We should add a footnote here that the ARVR system attempts to use these two ID's. For Rift/Index/MR this generally works as the controllers identify themselves as left and right and don't support additional controllers. But Vive wands do not. We do ensure in the latest version that trackers won't take slots 1 and 2.

**Comment by BastiaanOlij** [comment]
@TwistedTwigleg you do need to fix the merge conflict :)

**Comment by TwistedTwigleg** [comment]
Thanks for looking over it @BastiaanOlij! I'll try to make the suggested changes and fix the merge conflict once I have the time. Right now I think I might be able to squeeze it in tomorrow if nothing unexpected comes up.
Thanks again!

**Comment by TwistedTwigleg** [comment]
I fixed the merge conflict and made changes based on feedback. It should be good to go now 👍

**Comment by akien-mga** [comment]
Thanks a lot!

**Pull request merged**